config/src/iroha.rs

```diff
@@ -88,48 +87,56 @@ impl ConfigurationProxy {
-                        .to_owned()
-                ))
+                return Err(ConfigError::MissingField {
+                    field: "PUBLIC_KEY and PRIVATE_KEY",
+                    message: "The sumeragi keypair is not provided in the example configuration. It's done this way to ensure you don't re-use the example keys in production, and know how to generate new keys. Please have a look at \n\nhttps://hyperledger.github.io/iroha-2-docs/guide/configure/keys.html\n\nto learn more.\n\n-----",
+                });
             }
             // Second, torii gateway and sumeragi peer id are interchangeable too; the latter is derived from the
             // former and overwritten silently in case of difference
             if let Some(torii_proxy) = &mut self.torii {
                 if sumeragi_proxy.peer_id.is_none() {
                     sumeragi_proxy.peer_id = Some(iroha_data_model::peer::Id::new(
-                        &torii_proxy.p2p_addr.clone().ok_or_else(|| {
-                            eyre::eyre!("Torii `p2p_addr` field has `None` value")
-                        })?,
+                        &torii_proxy
+                            .p2p_addr
+                            .clone()
+                            .ok_or(ConfigError::MissingField {
+                                field: "p2p_addr",
+                                message:
+                                    "`p2p_addr` should not be set to `null` or `None` explicitly.",
+                            })?,
                         &self.public_key.clone().expect(
                             "Iroha `public_key` should have been initialized above at the latest",
                         ),
                     ));
                 } else {
                     // TODO: should we just warn the user that this value will be ignored?
-                    eyre::bail!(ConfigError::ProxyBuildError(
-                        "Sumeragi should not be provided with `peer_id` directly. It is computed from the other provided values.".to_owned()
-                    ))
+                    // TODO: Consider eliminating this value from the public API.
+                    return Err(ConfigError::ProvidedInferredField {
+                        field: "PEER_ID",
+                        message: "The `peer_id` is computed from the key and address. You should remove it from the config.",
+                    });
                 }
             } else {
-                eyre::bail!(ConfigError::ProxyBuildError(
-                    "Torii config should have at least `p2p_addr` provided for sumeragi finalisation"
-                        .to_owned()
-                ))
+                return Err(ConfigError::MissingField{
+                    field: "p2p_addr",
+					message: "Torii config should have at least `p2p_addr` provided for sumeragi finalisation",
```

Comment:
you have tabs in a few places

Reply:
I'm guessing `rustfmt` isn't replacing them with spaces.